.github/renovate.json5

```diff
@@ -0,0 +1,106 @@
+  labels: ["dependencies"],  // For convenient searching in GitHub
+  pip_requirements: {
+    fileMatch: ["^tox.ini$", "(^|/)requirements([\\w-]*)\\.txt$"]
+  },
+  pip_setup: {
+    fileMatch: ["^pyproject.toml$", "(^|/)setup\\.py$"]
+  },
+   packageRules: [
+    {
+      // Automerge patches, pin changes and digest changes.
+      // Also groups these changes together.
+      groupName: "bugfixes",
+      excludePackageNames: ["ruff"],
+      matchUpdateTypes: ["patch", "pin", "digest"],
+      prPriority: 3, // Patches should go first!
+      automerge: true
+    },
+    {
+      // Update all internal packages in one higher-priority PR
+      groupName: "internal packages",
+      matchPackagePrefixes: ["craft-", "snap-"],
+      matchLanguages: ["python"],
+      prPriority: 2
+    },
+    {
+      // GitHub Actions are higher priority to update than most dependencies.
+      groupName: "GitHub Actions",
+      matchManagers: ["github-actions"],
+      prPriority: 1
+    },
+    // Everything not in one of these rules gets priority 0 and falls here.
+    {
+      // Minor changes can be grouped and automerged for dev dependencies, but are also deprioritised.
+      groupName: "development dependencies (minor and patch)",
+      groupSlug: "dev-dependencies",
+      matchPackageNames: [
+        "black",
+        "codespell",
+        "coverage",
+        "isort",
+        "mypy",
+        "pydocstyle",
+        "pylint",
+        "pytest",
+        "tox",
+      ],
+      matchPackagePatterns: [
+        ".*-mock$"
+      ],
+      matchPackagePrefixes: [
+        "pytest-",
+        "pylint-",
+        "types-",
+      ],
+      matchUpdateTypes: ["minor", "patch", "pin", "digest"],
+      prPriority: -1,
+      automerge: true
+    },
+    {
+      // Documentation related updates
+      groupName: "documentation dependencies",
+      groupSlug: "doc-dependencies",
+      matchPackageNames: ["Sphinx"],
+      matchPackagePatterns: ["^[Ss]phinx.*$", "^furo$"]
+    },
+    {
+      // Other major dependencies get deprioritised below minor dev dependencies.
+      matchUpdateTypes: ["major"],
+      prPriority: -2
+    },
+    {
+      // Major dev dependencies are stone last, but grouped.
+      groupName: "development dependencies (major versions)",
+      groupSlug: "dev-dependencies",
+      matchDepTypes: ["devDependencies"],
+      matchUpdateTypes: ["major"],
+      prPriority: -3
+    },
+    {
+      // Ruff is still unstable, so update it separately.
+      matchPackageNames: ["ruff"],
+      prPriority: -3
+    }
+  ],
+  regexManagers: [
+    {
+      // tox.ini can get updates too if we specify for each package.
+      fileMatch: ["tox.ini"],
+      depTypeTemplate: "devDependencies",
+      matchStrings: [
+        "# renovate: datasource=(?<datasource>\\S+)\n\\s+(?<depName>.*?)(\\[[\\w]*\\])*[=><]=?(?<currentValue>.*?)\n"
+      ]
+    }
+  ],
+  timezone: "Etc/UTC",
+  automergeSchedule: "every weekend",
```

Comment:
sounds a bit dangerous to set this option right off the bat. I'd instead tryout Renovate in this project for a couple of weeks 1st, without automerge, and then enable this. Just to be sure

Reply:
fwiw, this only applies to the in development branch and wouldn't merge if the tests don't pass.

Reply:
fair enough

Reply:
I do think that's a reasonable concern for above (lines 19 and 60 enable automerge for specific groups if the tests succeed). I've made some improvements to the base renovate configuration that @mr-cal is reviewing and that I'd like to bring in here, and when I do so I'll comment out the automerge lines for rockcraft and set up a reminder to discuss it later.

Reply:
I've brought in the latest updates and commented out the automerges, so until we change that it'll only ever create PRs for us to merge.